prefect_snowflake/database.py

```diff
@@ -566,7 +580,7 @@ async def snowflake_query(
     query: str,
     snowflake_connector: SnowflakeConnector,
     params: Union[Tuple[Any], Dict[str, Any]] = None,
-    cursor_type: SnowflakeCursor = SnowflakeCursor,
+    cursor_class: Optional[type[SnowflakeCursor]] = SnowflakeCursor,
```

Comment:
Also, I think `Optional` should only be used if the default is `None`
```suggestion
    cursor_type: type[SnowflakeCursor] = SnowflakeCursor,
```